src/models/group.js

```diff
@@ -1,19 +1,25 @@
 import mongoose from 'mongoose';
-import User from './user';
+import uuid from 'uuid';
 
 const modelName = 'Group';
 
 const groupSchema = new mongoose.Schema({
   inviteCode: { type: String, required: true, unique: true },
-  users: [User.schema],
-  creator: { type: User.schema, required: true },
-  invites: [User.schema],
+  users: [mongoose.Schema.Types.ObjectId],
+  creator: { type: mongoose.Schema.Types.ObjectId, required: true },
+  invites: [mongoose.Schema.Types.ObjectId],
```

Comment:
Invites can be embedded directly, as opposed to referenced in the document. This is because they're only used in *one specific* group, it's a one-to-one relationship and isn't referenced in other places besides the group that owns the invite.

Reply:
Frankly, I am not too sure on what invites is even supposed to hold. This was added in Dimas PR

Reply:
**Invites** are the **pending** users, while **users** are approved users, which can use the group

Reply:
> **Invites** are the _pending_ users, while **users** are _approved_ users, which can use the group

I believe what Armin's was asking was: what was supposed to be stored *within* an invite.

Reply:
I don't think that you are referencing schemas correctly, from what I have seen it should be `{ type: mongoose.Schema.Types.ObjectId, ref: 'User'`

Reply:
> I don't think that you are referencing schemas correctly, from what I have seen it should be `{ type: mongoose.Schema.Types.ObjectId, ref: 'User'`

It doesn't need to be stored as a reference at all

Reply:
> > I don't think that you are referencing schemas correctly, from what I have seen it should be `{ type: mongoose.Schema.Types.ObjectId, ref: 'User'`
> 
> It doesn't need to be stored as a reference at all

Oh, you have already mentioned in another conversation

Reply:
So then should it just be an array of Users? (`invites: [User]`).

Reply:
> So then should it just be an array of Users? (`invites: [User]`).

1. I still didn't get the difference between referencing and the actual array of schemas.
2. It was supposed to be an array of Users in my idea of implementation, but we can change it if we have a better idea